src/Infrastructure/LeanCode.Serialization/JsonConverters/JsonLaxDateOnlyConverter.cs

```diff
@@ -1,16 +1,26 @@
+using System.Buffers;
 using System.Globalization;
+using System.Text;
 using System.Text.Json;
 using System.Text.Json.Serialization;
 
 namespace LeanCode.Serialization;
 
 public class JsonLaxDateOnlyConverter : JsonConverter<DateOnly>
 {
+    private static readonly UTF8Encoding encoder = new(false);
+
     public override DateOnly Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
     {
-        return reader.GetString() is string s
-            ? DateOnly.FromDateTime(DateTime.Parse(s, CultureInfo.InvariantCulture))
-            : default;
+        var value = reader.ValueSpan;
+        Span<char> chars = stackalloc char[value.Length];
+        encoder.GetChars(value, chars);
+        if (DateTime.TryParse(chars ,out DateTime date))
+        {
+            return new DateOnly(date.Year, date.Month, date.Day);
+        }
+
+        throw new JsonException("Invalid date format");
     }
 
     public override void Write(Utf8JsonWriter writer, DateOnly value, JsonSerializerOptions options) =>
```

Comment:
While you're at it, let's update this method to write to a span instead (see my comment in the other converter).